src/averageAges.js

```diff
@@ -37,7 +56,26 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womanCheck = person => person.sex === 'f';
+  const motherCheck = motherName => {
+    const mother = people.find(person => person.mother === motherName);
+
+    return mother;
+  };
+
+  let women;
+
+  if (arguments.length === 2) {
+    women = people.filter(person => motherCheck(person.name));
+  } else {
+    women = people.filter(womanCheck);
+  }
+
+  const womenAges = women.map(woman => woman.died - woman.born);
+  const sumAges = womenAges.reduce((sum, age) => sum + age);
+  const averageAge = sumAges / womenAges.length;
```

Comment:
You have the same logic for both women and men. Let's create a separate function for calculating an average age of the array